### ISSUE TITLE
Issue management: organize labels and milestones

### ISSUE THREAD

**Issue by mourner** [Leaflet/Leaflet#3443]
Currently Leaflet issue management with current labels and milestones is a bit of a mess. I tried to introduce some system in the past a couple times but always failed to keep up. 

We need a strict way to label issues so that any incoming issue can be labeled straight away and then we can sort through labels easily. Then we would describe the system in the contributing guide for other contributors to use to help me with issue management.

This means coming up with a list of the most useful labels, deleting some labels, converting some milestones into labels, and labeling all of our issues without exception.

Examples of labels we could depend on:
- `needs-more-info` (incomplete cases, no JSFiddle etc.)
- `needs-discussion` (can't decide about something)
- `help-wanted` (issues that are not hard to do and where contributions would be welcome)
- `accepted` (core devs acknowledged that this needs to be fixed)

Lets discuss! What other labels are useful? What labels only add clutter and should be removed?

cc @Leaflet/leaflet-core-contributors

**Comment by yohanboniface** [comment]
Maybe `design-decision-needed`, to mark that an issue needs your special attention on top of the team one?

Also, I suggest to replace `later` by `future`.

**Comment by perliedman** [comment]
Yes, this sounds good. Especially making this a bit more formal is a good idea, to make it easier to add correct labels; I don't think I've personally set a label for any issue before.

It's a good thing to have labels for different technical subjects, for example I sometimes filter for issues with the `projections` label. At the same time, it gets cluttered if they are too fine grained. For example, I'd vote for removing `max-bounds`, which sounds like something that could perhaps be replaced by a coordinating issue if there are several issues related to this.

**Comment by yohanboniface** [comment]
> For example, I'd vote for removing max-bounds

+1, I was about to suggest the same ;)

**Comment by mourner** [comment]
I added `maxbounds` label at some point because there was a lot of related issues — the feature was too buggy. :) Not that it stabilized somewhat (3 open issues left), I'm removing it.

**Comment by mourner** [comment]
BTW we just hit 2000 closed issues and 1200 closed pull requests, so the problem is very important. :)

**Comment by IvanSanchez** [comment]
I'd like to see more workflow-oriented labels. e.g. a `tick-tock` label meaning "if no-one replies to this, we'll close it in two months".

**Comment by mourner** [comment]
I'd like to keep things simple, with self-descriptive labels. :) E.g. in the case above you'd use `needs-more-info` for the same purpose.

**Comment by IvanSanchez** [comment]
In any case, I'd put the relationships between labels and workflow somewhere in `CONTRIBUTING.md`.

**Comment by mourner** [comment]
Yeah, I'm now fiddling with different labels trying to see what works well and what doesn't. I'll write a comment summarizing the labeling system with descriptions for everyone to discuss, and then we'll move this to `CONTRIBUTING.md`.

**Comment by mourner** [comment]
OK, here's an initial list of labels I suggest to use consistently. Corrections and comments are very welcome!

### Issue category

- `feature` — new functionality
- `bug` — something that should be fixed
- `docs` — anything related to docs and the Leaflet website

### Workflow

These labels would be used in something like [waffle.io](https://waffle.io/) as columns to help sort through the issue queue.

- `accepted` — project maintainers acknowledged that this issue needs to be addressed ("backlog")
- `ready` — the issue is ready to be worked on (reproducible, there are clear steps to fix/implement, etc.), similar to "next actions"
- `in progress` — the issue is being worked on by someone; also automatically labeled by Waffle when someone opens a PR referencing the issue
- `needs review` – incoming PRs by non-collaborators, automatically labeled by Waffle
- `later` — project maintainers read the issue but didn't yet decide whether it should be addressed and how, and put it aside for now

### Additional labels

- `needs discussion` — more discussion and feedback is needed before we can proceed on the issue
- `needs more info` — any issue where we expect more comments by people participating in the issue, e.g. waiting for JSFiddle or more details
- `needs solution` — any issue we have no idea how to solve technically, so it's on hold; help wanted
- `help wanted` — any issue that we'd like help on; this includes issues that we'd like to fix but don't have the bandwidth for right now, inviting contributors to look at them
- `easy fix` — any issue that's easy to fix, just takes time to test/etc., so it's a good entry point for new contributors
- `blocker` — any serious issue that breaks Leaflet maps for a group of people and should be prioritized

### Issue topics

- `performance`
- `api` — changes to classes, methods and events
- `android` — dreaded Android bugs
- `mobile` — anything related to mobile devices
- `projections`
- `refactoring` — mostly internal changes that don't affect the user-facing API or UX much, but are necessary
— `ux` — user experience (interaction features, controls, popups, visual appearance etc.)

**Comment by mourner** [comment]
Waffle board to help sort through the issues: https://waffle.io/leaflet/leaflet

**Comment by mourner** [comment]
I should probably rename `ready` to `next` to make it clear. Then I'd just use the label to queue issues we're going to work on next (Waffle supports sorting manually with drag-n-drop).

**Comment by mourner** [comment]
Also maybe there should also be an `investigate` label which indicates that the issue has all the necessary info (JSFiddle, etc.), but needs to be investigated further to confirm that it's actually a Leaflet issue (something in between `needs-more-info` and `accepted`).

**Comment by yohanboniface** [comment]
> I should probably rename ready to next to make it clear

`next` doesn't sound clearer to me, what about `confirmed`?

But I like the overall scheme :)

**Comment by danzel** [comment]
I'm not totally sure the difference between ```accepted``` and ```ready```. Should we combine them?
```confirmed``` seems to be what most projects use.

**Comment by mourner** [comment]
`accepted` is the same as `confirmed`, it just has a bit broader meaning as the word makes sense on feature requests, while "confirmed" is usually only applied to bug reports. Both are pretty common.

The `ready` (or `next`) label is needed to pick something from `accepted` (where there would be hundreds of issues), and mark it as "I'm going to work on this soon", putting in an appropriate place among other immediate next tasks.

I didn't yet sort through all the open issues properly, but would like to do this in the next couple days.

@danzel happy to see you commenting! You don't seem very active on GH recently.

**Comment by danzel** [comment]
So ```ready``` is ```accepted``` once it is assigned? ok.

Yeah been waaaay to busy with other things :(

**Comment by mourner** [comment]
Yeah, once it's assigned _and_ you're committed to work on it in near future.

**Comment by mourner** [comment]
Looks like this label system is working pretty well. Leaflet issues feel much more organized. So lets document in CONTRIBUTING.md and close the issue.

**Comment by FadilEldin** [comment]
Should be closed
